### DISCUSSION TITLE
tokio_stream WatchStream isn't a drop-in replacement for Tokio 0.2 watch::Receiver

### DISCUSSION THREAD

**Discussion by krallin** [tokio-rs/tokio#3575] [General] [answered]
In Tokio 0.2, watch::Receiver is a stream, and it always yields the initial value. This is convenient when e.g. implementing a background task that is allowed to fall behind and should just operate on the last value.

In Tokio 1.x, watch::Receiver isn't a stream and you have to use `tokio_stream::wrappers::WatchStream` instead. The only problem is that isn't a drop-in replacement, because that stream doesn't yield the first value. This isn't this really obvious from the docs at all, and it also feels pretty awkward to provide a default value that is never used.

A couple questions:

- Is this intentional? If not, should this be changed? After all, tokio_steam hasn't hit 1.x.
- if it is intentional, perhaps this is worth documenting?

**Comment by Darksonn** [comment, marked as answer]
No, this isn't intentional.

**Reply by krallin** [reply]
Glad to hear it :) I'll prep a PR to fix this then if that works for you?

**Reply by Darksonn** [reply]
Go ahead.
